### ISSUE TITLE
Handshake error when trying to use an emitter and a listener on a connection

### ISSUE THREAD

**Issue by KevinGrandon** [mtth/avsc#70]
We have some code that looks like this:
```
const protocol = avsc.parse(attrs);

const net = require('net');
net.createServer()
  .on('connection', (con) => {
    const listener = protocol.createListener(con);
    const emitter = protocol.createEmitter(con);
  })
  .listen(port);
```

Trying to create an emitter after calling createListener seems to throw a handshake error:

```
Uncaught Exception:
Error: handshake error: invalid org.apache.avro.ipc.HandshakeMatch enum index: 3631
    at wrapError (project_path/node_modules/avsc/lib/protocols.js:1409:13)
    at NettyDecoder.onPing (project_path/node_modules/avsc/lib/protocols.js:628:26)
    at emitOne (events.js:96:13)
    at NettyDecoder.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at NettyDecoder.Readable.push (_stream_readable.js:134:10)
    at NettyDecoder.Transform.push (_stream_transform.js:128:32)
    at NettyDecoder._transform (project_path/node_modules/avsc/lib/protocols.js:1326:12)
    at NettyDecoder.Transform._read (_stream_transform.js:167:10)
    at NettyDecoder.Transform._write (_stream_transform.js:155:12)
```

Thank you for the great library. We'd love to see this work or if not, possibly a more useful error message so we know how to fix this.

**Comment by mtth** [comment]
It's currently not possible to use the same stream to create more than one emitter or listener. Would you be able to use a single emitter/listener or spread them over multiple connections? Could you tell me more about your use-case?

The error could definitely be more helpful, I'll improve the message.

**Comment by mtth** [comment]
After thinking a bit about it, it didn't seem too difficult to support multiplexing: I just added a new option to enable it. Any number of emitters and listeners can now share the same connection (or any stateful transport) as long as they have distinct `scope`s  (see [`protocol.createEmitter`](https://github.com/mtth/avsc/wiki/API#protocolcreateemittertransport-opts) and [`protocol.createListener`](https://github.com/mtth/avsc/wiki/API#protocolcreatelistenertransport-opts)).

Here's how you would use it in your example (just make sure that you also update the client code accordingly, the scopes should match between an emitter/listener pair):

```javascript
const protocol = avsc.parse(attrs);

const net = require('net');
net.createServer()
  .on('connection', (con) => {
    const listener = protocol.createListener(con); // Default scope.
    const emitter = protocol.createEmitter(con, {scope: 'push'}); // Different scope.
  })
  .listen(port);
```

The option is available in `4.1.8`. Let me know if this doesn't fully cover your use-case.

**Comment by KevinGrandon** [comment]
@mtth - That's awesome, the latest version seems to be working well in my tests so far. Nice work, and thanks for the quick support here :)
